packages/active-campaign-client/src/handlers/manageListSubscription.ts

```diff
@@ -0,0 +1,40 @@
+import { APIGatewayProxyResult } from 'aws-lambda';
+import { acClient } from '../utils/activeCampaignClient';
+
+export async function manageListSubscription(
+  username: string,
+  listName: string,
+  action: 'subscribe' | 'unsubscribe'
+): Promise<APIGatewayProxyResult> {
+  try {
+    const contactId = await acClient.getContactByCognitoId(username);
+    if (!contactId) {
+      return {
+        statusCode: 404,
+        body: JSON.stringify({ message: 'Contact not found' }),
+      };
+    }
+
+    const listId = await acClient.getListIdByName(listName);
```

Comment:
What if the `listId` does not exist?

Reply:
`getListIdByName` throws an exception that it is catched and returned the error